src/main/kotlin/com/example/flashcardbackend/deckgroup/DeckGroupRepository.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.flashcardbackend.deckgroup
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.jdbc.core.RowMapper
+import org.springframework.stereotype.Repository
+import org.springframework.transaction.annotation.Transactional
+import java.sql.ResultSet
+import java.sql.SQLException
+
+class DeckGroupRowMapper : RowMapper<DeckGroupListItemDTO> {
+    @Throws(SQLException::class)
+    override fun mapRow(resultSet: ResultSet, i: Int): DeckGroupListItemDTO {
+        return DeckGroupListItemDTO(
+            resultSet.getInt("id"),
+            resultSet.getString("collection_name"),
+            resultSet.getString("description"),
+        )
+    }
+}
+
+@Transactional
+@Repository
+class DeckGroupRepository(@Autowired val jdbcTemplate: JdbcTemplate) {
+
+    fun findAll(): List<DeckGroupListItemDTO> {
```

Comment:
This should return list of domain objects (maybe DeckGroupListItem?). This way your repository API stays consistent (domain object in, domain object out).

Reply:
thanks for pointing it out! yeah, forgot to take care of the way out (domain object out). should have checked the code again before PR.